### PR TITLE
Add support for k8s Resource metrics as source of scaling

### PR DESCRIPTION
**Pull request by azun**
## Description

Add support for k8s Resource metrics as source of scaling

## How Has This Been Tested?

mage operator:tests passes all tests
Tested in k8s using a nginx deployment with the config added in examples/resource-scaler.yaml

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by amuraru** [comment]
@azun can you share a bit more details on why this is needeD?

File: replicas/replicas_calculator.go [hunk 1]
```diff
@@ -38,10 +38,16 @@ func (rc *ReplicaCalculator) CalculateReplicas(currentReplicas int32, metricTarg
 		return rc.calculateValue(currentReplicas, metricTarget, metricValues)
 	case v1alpha1.AverageValueMetricType:
 		return rc.calculateAverageValue(currentReplicas, metricTarget, metricValues)
+		// case v1alpha1.UtilizationMetricType:
```
Comment (alexander-falca) [inline replicas/replicas_calculator.go:41]: any reasons for commenting them out?

**Review by alexander-falca — APPROVED** [review]
Minor comment and then :+1:

File: replicas/replicas_calculator.go (hunk 1 above)
Comment (azun) [reply replicas/replicas_calculator.go:41]: They got mixed with a different patch that implements them. I'll clean it up

**Pull request merged**